os/fs/smartfs/smartfs_smart.c

```diff
@@ -810,7 +810,14 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 			/* Now get the chained sector info and reset the offset */
 
 			sf->curroffset = sizeof(struct smartfs_chain_header_s);
-			sf->currsector = SMARTFS_NEXTSECTOR(header);
+
+			/* If file modified in this loop but buf if buf data is remain, it will try to append data.
```

Comment:
Do you mean >
 If file modified in this loop but if buf is not empty , it will try to append data ?

Reply:
yes it will get nextsector and go to outside of loop.

Reply:
I understand what you are trying to convey with the comment.

Could you please change the comment as follows: -
> If file is modified and more data remains to be appended to file, but no next sector is available,
do not update sf->currsector and curroffset,
Will be handled when buffer data is synced.

Reply:
oh. ok thanks!